oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -618,26 +606,35 @@ impl Runtime {
         // for any additional work to be finished here. This may take an arbitrary amount of time,
         // depending on the work that the Node thread has to perform, but at least we know that the
         // it will not be able to enter again in a blocking state.
-        for (node_id, instance) in self
-            .node_join_handles
-            .lock()
-            .expect("could not acquire lock on node_join_handles")
-            .drain()
-        {
-            debug!("join thread for {:?}...", node_id);
-            if let Err(err) = instance.join() {
-                error!("could not terminate node: {:?}", err);
+        let join_handles = self.take_join_handles();
+        for (node_id, join_handle_opt) in join_handles {
+            if let Some(join_handle) = join_handle_opt {
+                debug!("join thread for {:?} ...", node_id);
+                if let Err(err) = join_handle.join() {
```

Comment:
Will it stop a Node that is currently blocked on waiting on the TCP socket?

Reply:
I don't know, but I don't think this PR would change that behaviour.  Or is there some existing trigger that I've missed?

Reply:
No, I think everything will be the same

Reply:
AFAICT `join` just waits for the thread to naturally terminate, it does not really signal anything at all, the other thread would not even know it is been waited on.